tests/test_io_layer_distribution.c

```diff
@@ -128,13 +128,27 @@ static void test_distribute_over_two_layers(void) {
   cleanup_layers(&two_layers);
 }
 
+static void test_normalize_profile() {
```

Comment:
With more time to spend I would suggest to add a failing test as well. Comment can be closed after reading.